src/text-field/README.md

```diff
@@ -147,3 +147,34 @@ An icon can be added to the right of the text field by adding an element with th
   <span slot="end" class="codicon codicon-chevron-right"></span>
 </vscode-text-field>
 ```
+
+### Slotted Content
+
+Building on top of the icon examples, it's worth noting that any arbitrary HTML can be inserted into the start and end slots of a text field with the `slot="start"` and `slot="end"` attributes.
+
+The below example demonstrate how the native search text field from VS Code might be _visually_ implemented with a `section` tag that wraps a few `vscode-buttons` with an icon appearance applied (please note that further JavaScript is needed to make this example functional).
+
+**❗️ An important note ❗️**
+
+While component slots provide a lot of flexibility to extend components please use them responsibly and remember to follow the extension [UX Guidelines](https://code.visualstudio.com/api/ux-guidelines/overview) to ensure that your extension UI still has the same look and feel as the rest of VS Code.
```

Comment:
Might be worth just calling out what VS Code actually puts into text fields as an example. And optionally some examples of what not to do? I don't think the UX guidelines will really cover this specific of a case.